moler/helpers.py

```diff
@@ -574,3 +574,92 @@ def regexp_without_anchors(regexp):
     if regexp_str == org_regexp_str:
         return regexp
     return re.compile(regexp_str)
+
+
+def remove_state_from_sm(source_sm: dict, source_transitions: dict, state_to_remove: str) -> tuple:
+    """
+    Remove a state from a state machine dict.
+    :param source_sm: a dict with a state machine description
+    :param source_transitions: a dict with a state machine transitions
+    :param state_to_remove: name of state to remove
+    :return: tuple with 2 dicts without state_to_remove, 0 - new state machine, 1 - new transitions
+    """
+    from pprint import pformat
+    print(f"source sm: {pformat(source_sm)}")
+    print(f"source transitions: {pformat(source_transitions)}")
+    new_sm = copy.deepcopy(source_sm)
+    new_transitions = copy.deepcopy(source_transitions)
+
+    from_states = []
+    for from_state in source_sm.keys():
+        for to_state in source_sm[from_state].keys():
+            if to_state == state_to_remove:
+                from_states.append(from_state)
+
+    for to_state in source_sm[state_to_remove].keys():
+        if to_state == state_to_remove:
+            continue
+        for new_from in from_states:
+            if new_from != to_state and new_from != state_to_remove:
```

Comment:
is it necessary? ` new_from != state_to_remove` I mean, is it possible situation that  ` new_from == state_to_remove`?

Reply:
You're right 👍

Reply:
done